src/declarativewebutils.cpp

```diff
@@ -75,10 +81,35 @@ int DeclarativeWebUtils::getLightness(QColor color) const
     return color.lightness();
 }
 
-bool DeclarativeWebUtils::fileExists(QString fileName) const
+QString DeclarativeWebUtils::getUniquePictureName(const QString& fileName) const
 {
-    QFile file(fileName);
-    return file.exists();
+    int collisionCount(0);
+    const QString picturesDir(BrowserPaths::picturesLocation());
```

Comment:
Wrote this out of my head but it should depict the idea. Rather than mangling the file manually utilize QFileInfo instead.

```
...
QString result = picturesDir + "/" + fileName;
QFileInfo fileInfo(fileName);
// Path, base name, count, and suffix
QString newFile("%1/%2(%3).%4");
QString baseName = fileInfo.baseName();
QString suffix = fileInfo.completeSuffix();
while (QFileInfo::exists(result)) {
    collisionCount++;
    result = newFile.arg(picturesDir).arg(baseName).arg(collisionCount).arg(suffix);
}
...
```